source/ada/lsp-ada_driver.adb

```diff
@@ -58,5 +66,28 @@ begin
       Handler'Unchecked_Access,
       Handler'Unchecked_Access);
 
-   Server.Run;
+   loop
+      begin
+         Server.Run;
+      exception
+         when E : Property_Error =>
+            Server_Trace.Trace
+              ("LAL Property Error:" & Exception_Message (E));
+            Server_Trace.Trace (Symbolic_Traceback (E));
+
+         when Ada.IO_Exceptions.End_Error =>
+            Server_Trace.Trace
```

Comment:
I would prefer Failure exit status here. One of the test in our tiny testsuite checks this.

Reply:
You mean rather than checking for End_Error ? This was the shortest path, I modified the function that raised a Constraint_Error before. We can further modify it to use  error codes instead, but I would suggest doing that in a second path.

Reply:
Ah ok, I think I understood what you meant. You want the server to quit with an error code on this condition right ? In that case: OK, will take care of it !

FWIW: VS Code always quit like that on my system for some reason, which is why I didn't assume that it was an error case ...